Consolidated project settings into settings UI

eightants opened the pull request:
## Summary

Continuing work on new settings UI. 
- Consolidated project settings onto this page, multiple projects
- Removed top right buttons from settings ui
- Navbar only has the Back to Project button for all settings pages, including project settings. Initially the project settings page had Sessions, Errors, Logs, and all other settings on the navbar. Removed them to keep it consistent which I think makes sense?

https://github.com/highlight/highlight/issues/4917

## How did you test this change?

Clicking around and changing settings to make sure they work. Also, checked that URLs are preserved.

<img width="1440" alt="Screenshot 2023-04-25 at 12 31 13 PM" src="https://user-images.githubusercontent.com/29858539/234342875-a50c1c35-f50d-441a-97f4-8b88473b35e4.png">

## Are there any deployment considerations?

NA

Vadman97 commented on frontend/src/pages/SettingsRouter/SettingsRouter.tsx:
```diff
@@ -180,6 +213,27 @@ export const SettingsRouter = () => {
+							<Text size="xxSmall" color="secondaryContentText">
+								Project Settings
+							</Text>
+						</Box>
+						{projectSettingTabs.map((project) => (
+							<NavLink
+								key={project.key}
+								to={`/${project.key}/settings/recording`}
+								className={clsx(styles.menuItem, {
+									[styles.menuItemActive]:
+										projectId === project.key,
+								})}
+							>
+								<Stack direction="row" align="center" gap="4">
+									<Text>{project.title}</Text>
+								</Stack>
+							</NavLink>
+						))}
+					</Stack>
```
in the [figma](https://www.figma.com/file/rdBPNzn7Klk6RG1LHUCE5B/Screen-Designs?type=design&node-id=7552%3A287169&t=CUu6aswgUI0QCnpi-1), this also has a color button on the side. could we add that?

eightants replied:
What color should it be? Just some random color?

Vadman97 replied:
could we use the project name hashed to an int to look up from our color pallet? similar to the avatar code https://github.com/highlight/highlight/blob/09c50163505c36bcdeaaed265722c4cd0cc684a1/frontend/src/components/Avatar/Avatar.tsx#L78

Vadman97 commented on frontend/src/pages/SettingsRouter/SettingsRouter.tsx:
```diff
@@ -180,6 +213,27 @@ export const SettingsRouter = () => {
 						))}
 					</Stack>
+					<Stack gap="0">
+						<Box mt="12" mb="10" ml="8">
```
@jay-khatri @julian-highlight does it make sense to have the left panel organized as `Account Settings`, `Workspace Settings`, `Project Settings` instead? more of a heirarchy of what belongs to what

deltaepsilon commented on frontend/src/pages/SettingsRouter/SettingsRouter.tsx:
```diff
@@ -31,7 +33,19 @@ export type WorkspaceSettingsTab =
 	| 'upgrade-plan'
-	| SettingGroups
+
+export const isSettingsPath = (path: string[]) => {
```
This always takes in a `path` value from `useLocation`, so it can easily be refactored into a custom hook to better encapsulate all of it's dependent logic.

deltaepsilon commented on frontend/src/pages/SettingsRouter/SettingsRouter.tsx:
```diff
@@ -117,6 +143,13 @@ export const SettingsRouter = () => {
 		],
 	]
 
+	const projectSettingTabs = allProjects
```
**non-blocking**

These sorts of functions can be wrapped with a `useMemo`. The primary reason for `useMemo` is to maintain stable references. The secondary reason is to cut down on expensive operations.

This particular operation isn't particularly expensive, so it's up to you.

eightants commented:
Addressed the comments and added the sidebar border from https://github.com/highlight/highlight/pull/5097

Will merge this Monday, since there wasn't any blocking changes when approved last week.
